packages/tasks/src/gguf.ts

```diff
@@ -1,3 +1,155 @@
+// This list is copied from gguf/types.ts, but will all types available (for backward compatibility)
+// NOT to be confused with GGMLQuantizationType, a FileQuantization can contain multiple GGMLQuantizationType
+// For example, Q4_K_M model can contains Q4_K and Q6_K tensors
+export enum GGMLFileQuantizationType {
+	F32 = 0,
+	F16 = 1,
+	Q4_0 = 2,
+	Q4_1 = 3,
+	Q4_1_SOME_F16 = 4,
+	Q4_2 = 5,
+	Q4_3 = 6,
+	Q8_0 = 7,
+	Q5_0 = 8,
+	Q5_1 = 9,
+	Q2_K = 10,
+	Q3_K_S = 11,
+	Q3_K_M = 12,
+	Q3_K_L = 13,
+	Q4_K_S = 14,
+	Q4_K_M = 15,
+	Q5_K_S = 16,
+	Q5_K_M = 17,
+	Q6_K = 18,
+	IQ2_XXS = 19,
+	IQ2_XS = 20,
+	Q2_K_S = 21,
+	IQ3_XS = 22,
+	IQ3_XXS = 23,
+	IQ1_S = 24,
+	IQ4_NL = 25,
+	IQ3_S = 26,
+	IQ3_M = 27,
+	IQ2_S = 28,
+	IQ2_M = 29,
+	IQ4_XS = 30,
+	IQ1_M = 31,
+	BF16 = 32,
+	Q4_0_4_4 = 33,
+	Q4_0_4_8 = 34,
+	Q4_0_8_8 = 35,
+	TQ1_0 = 36,
+	TQ2_0 = 37,
+}
+
+const ggufQuants = Object.values(GGMLFileQuantizationType).filter((v): v is string => typeof v === "string");
+export const GGUF_QUANT_RE = new RegExp(`(?<quant>${ggufQuants.join("|")})` + "(_(?<sizeVariation>[A-Z]+))?");
+export const GGUF_QUANT_RE_GLOBAL = new RegExp(GGUF_QUANT_RE, "g");
+
+export function parseGGUFQuantLabel(fname: string): string | undefined {
+	const quantLabel = fname.toUpperCase().match(GGUF_QUANT_RE_GLOBAL)?.at(-1); // if there is multiple quant substrings in a name, we prefer the last one
+	return quantLabel;
+}
+
+// order of quantization, from biggest to smallest
+// this list must be in sync with the order in GGMLFileQuantizationType
+// the gguf.spec.ts tests are using verify if the order is correct
+export const GGUF_QUANT_ORDER: GGMLFileQuantizationType[] = [
```

Comment:
btw interested in improving the quant selector here: https://huggingface.co/unsloth/Qwen3-30B-A3B-GGUF?local-app=llama.cpp

Reply:
Yeah that's a good idea. This list is already exported and ready to be used in hub UI, do you think of any other improvements ?

Reply:
No I think we can start with it (maybe there's a few variants missing)

Reply:
I already sync'ed this list with latest llama.cpp code, so it should be good